### PR TITLE
[IMP] web: adapt the kanban "Remove" button label

### PR DESCRIPTION
**Pull request by thsh-odoo**
**Before this PR**

Remove button is shown when the view of our field is in kanban mode.

When our field is one2many it will delete the record and when our field is
many2one it will remove the record.

So many times user get confuse that they are deleting the record or removing
the record.

**After this PR**

So to clear out the confusion this commit changes the label of remove to
to delete if the field is one2many.

A test was added to ensure this ux change.

Task-3185967

### CODE REVIEW

**Comment by tde-banana-odoo** [comment]
@aab-odoo what do you think of this small pr ?

File: addons/web/static/src/views/fields/relational_utils.js [hunk 1]
```diff
@@ -564,6 +564,7 @@ X2ManyFieldDialog.props = {
     save: Function,
     title: String,
     delete: { optional: true },
+    button_text: String,
```
Comment (awa-odoo) [inline addons/web/static/src/views/fields/relational_utils.js:567]: Multiple things here:
- Don't use snake_case in js, we use camelCase.
- Can't we name it something more obvious? Maybe deleteButtonLabel ?
- I would make it optional and use the current value if not provided (aka "Remove") in the template.
- Not sure if we can add props in stable, definitely not required ones, but we will ask JS team to confirm after my review.

File: addons/web/static/src/views/fields/relational_utils.js [hunk 2]
```diff
@@ -639,6 +640,7 @@ export function useOpenX2ManyRecord({
         const form = await getFormViewInfo({ list, activeField, viewService, userService, env });
 
         let deleteRecord;
+        let button_text = "";
```
Comment (awa-odoo) [inline addons/web/static/src/views/fields/relational_utils.js:643]: Would set it to undefined and not empty text, it could possibly mean that we don't want a label is that case, which is strange.

File: addons/web/static/src/views/fields/relational_utils.js [hunk 3]
```diff
@@ -650,7 +652,8 @@ export function useOpenX2ManyRecord({
                 views: { form },
             });
             const { delete: canDelete, onDelete } = activeActions;
-            deleteRecord = viewMode === "kanban" && canDelete ? () => onDelete(_record) : null;
+            deleteRecord = viewMode === "kanban" && canDelete ? () => onDelete(_record) : null;            
+            button_text = activeActions.type === 'one2many' ? 'Delete' : 'Remove';
```
Comment (awa-odoo) [inline addons/web/static/src/views/fields/relational_utils.js:656]: You need to translate those.

File: addons/web/static/tests/views/fields/one2many_field_tests.js [hunk 4]
```diff
@@ -2510,6 +2510,39 @@ QUnit.module("Fields", (hooks) => {
         await clickSave(target);
     });
 
+    QUnit.test("open a record in a one2many kanban with should display delete button", async function (assert) {
```
Comment (awa-odoo) [inline addons/web/static/tests/views/fields/one2many_field_tests.js:2513]: Sugg:

> When viewing one2many records in an embedded kanban, the delete button should say "Delete" and not "Remove"

File: addons/web/static/tests/views/fields/one2many_field_tests.js [hunk 5]
```diff
@@ -2510,6 +2510,39 @@ QUnit.module("Fields", (hooks) => {
...
+                                <t t-name="kanban-box">
+                                    <div t-att-class="'oe_kanban_global_click'">
+                                        <h3>Record 1</h3>
+                                    </div>
+                                </t>
+                            </templates>
+                        </kanban>
+                        <form>
+                            <field name="turtle_foo"/>
+                        </form>
```
Comment (awa-odoo) [inline addons/web/static/tests/views/fields/one2many_field_tests.js:2535]: Why is there a form inside a form? This seems a bit unusual.

File: addons/web/static/tests/views/fields/one2many_field_tests.js [hunk 6]
```diff
@@ -2510,6 +2510,39 @@ QUnit.module("Fields", (hooks) => {
...
+                    </field>
+                </form>`,
+            resId: 1,
+        });
+
+        await click(target.querySelector(".o_kanban_record"));
```
Comment (awa-odoo) [inline addons/web/static/tests/views/fields/one2many_field_tests.js:2541]: Why do you need to click on the record?

File: addons/web/static/src/views/fields/relational_utils.js [hunk 7]
```diff
@@ -564,6 +564,7 @@ X2ManyFieldDialog.props = {
     save: Function,
```
Comment (awa-odoo) [inline addons/web/static/src/views/fields/relational_utils.js:564]: Can you rephrase the commit message, a bit like I suggested in the other PR?

One giant sentence is very hard to follow, split that into multiple sentences. 

I will also explain that it only changes the button label, right now when reading it feels like you are changing the behavior.

**Review by awa-odoo — COMMENTED** [review]
Hello @thsh-odoo 

A bit more work is required on this one.

Don't hesitate to ask if anything is unclear.

Cheers!

File: addons/web/static/tests/views/fields/one2many_field_tests.js (hunk 5 above)
Comment (thsh-odoo) [reply addons/web/static/tests/views/fields/one2many_field_tests.js:2535]: I have tried to create the similar view same as res.partner view in testcase
https://github.com/odoo/odoo/blob/16.0/odoo/addons/base/views/res_partner_views.xml#L235

File: addons/web/static/tests/views/fields/one2many_field_tests.js (hunk 6 above)
Comment (thsh-odoo) [reply addons/web/static/tests/views/fields/one2many_field_tests.js:2541]: Because after clicking the record we can see the footer buttons

File: addons/web/static/src/views/fields/relational_utils.js (hunk 1 above)
Comment (thsh-odoo) [reply addons/web/static/src/views/fields/relational_utils.js:567]: Hello @awa-odoo

Comment (awa-odoo) [reply addons/web/static/src/views/fields/relational_utils.js]: Hello.

I mean that as the prop is optional, you should check it in the xml template and do "t-if prop" -> print the prop label, "t-else", print the existing label (so "Remove").

Even if it's always specified in your use case, this component could be used in other contexts where the prop is not specified (as it's optional) and we will need a label for the button in that case.

Does that make sense?

Cheers!

Comment (thsh-odoo) [reply addons/web/static/src/views/fields/relational_utils.js]: Hello @awa-odoo 

Changes are done as per your suggestions can you please review it again :)

File: addons/web/static/src/views/fields/relational_utils.js (hunk 7 above)
Comment (awa-odoo) [reply addons/web/static/src/views/fields/relational_utils.js:564]: Looking better, thanks !

Can we just alter a bit the header as well?

Instead of:
> [FIX] web: ux change in field dialog footer

Would go for:
> [IMP] web: adapt the kanban "Remove" button label

**Review by awa-odoo — COMMENTED** [review]
Hello @thsh-odoo 

Looking a lot better already.
A few more comments and then we will ask the JS team for a final check and we will be good to go.

Thanks & cheers!

**Comment by thsh-odoo** [comment]
Hello @awa-odoo

Suggested changes are done.
Had to change the test case a little bit because it was giving traceback.
Can you review it again.

Thanks :)

**Comment by awa-odoo** [comment]
@robodoo r+
